src/Sql/Sql/ManagedInstanceDnsAlias/Cmdlet/GetAzureSqlManagedInstanceDnsAlias.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.ManagedInstance.Model;
+using Microsoft.Azure.Commands.Sql.ManagedInstanceDnsAlias.Model;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedInstanceDnsAlias.Cmdlet
+{
+    /// <summary>
+    /// Defines the Get-AzSqlServerDnsAlias cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlInstanceDnsAlias",
+        DefaultParameterSetName = GetByNameParameterSet),
+        OutputType(typeof(AzureSqlManagedInstanceDnsAliasModel))]
+    public class GetAzureSqlManagedInstanceDnsAlias : AzureSqlManagedInstanceDnsAliasCmdletBase
+    {
+        private const string GetByNameParameterSet = "GetByNameParameterSet";
+        private const string GetByParentObjectParameterSet = "GetByParentObjectParameterSet";
+        private const string GetByResourceIdParameterSet = "GetByResourceIdParameterSet";
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = GetByNameParameterSet, Position = 0, HelpMessage = "Name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of target managed instance.
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = GetByNameParameterSet, Position = 1, HelpMessage = "Name of the managed instance.")]
+        [ResourceNameCompleter("Microsoft.Sql/managedInstances", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        public string InstanceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the Azure Sql Managed Instance DNS Alias
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = GetByNameParameterSet, Position = 2, HelpMessage = "Name of the managed instance DNS alias.")]
+        [Parameter(Mandatory = false, ParameterSetName = GetByParentObjectParameterSet, Position = 1, HelpMessage = "Name of the managed instance DNS alias.")]
+        [Alias("DnsAliasName")]
+        [SupportsWildcards]
```

Comment:
how many aliases can there be per instance?

Reply:
I'm not sure if a limit exists, but there can be more than one